tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/AsynchronousCSTest.java

```diff
@@ -0,0 +1,327 @@
+        
+        CompletionStage<Connection> resultFuture = client.serviceCS(waitingFuture, true);
+        waitingFuture.completeExceptionally(new SimulatedException("completedExceptionally"));
+
+        Assert.assertTrue(isCompleted(resultFuture));
+        Assert.assertTrue(isCompletedExceptionally(resultFuture));
+        Assert.assertFalse(isCancelled(resultFuture));
+        Assert.assertThrows(SimulatedException.class, CompletableFutureHelper.toCompletableFuture(resultFuture)::get);
+
+        try {
+            waitUntilCompleted(resultFuture);
+        }
+        catch (CompletionException e) {
+            handleCompletionException(e);
+        }
+    }
+
+    /**
+     * Test that the stage returned by calling an asynchronous method is 
+     * completed exceptionally if the method returns a stage completed exceptionally
+     */
+    @Test
+    public void testAsyncCompletesExceptionallyWhenCompletedExceptionally() {
+        CompletableFuture<Void> waitingFuture = newWaitingFuture();
+        
+        CompletionStage<Connection> resultFuture = client.serviceCS(waitingFuture, false);
+        waitingFuture.completeExceptionally(new SimulatedException("completedExceptionally"));
+
+        Assert.assertTrue(isCompleted(resultFuture));
+        Assert.assertTrue(isCompletedExceptionally(resultFuture));
+        Assert.assertFalse(isCancelled(resultFuture));
+        Assert.assertThrows(SimulatedException.class, CompletableFutureHelper.toCompletableFuture(resultFuture)::get);
+
+        try {
+            waitUntilCompleted(resultFuture);
+        }
+        catch (CompletionException e) {
+            handleCompletionException(e);
+        }
+    }
+
+    /**
+     * Ensure that any waiting futures get completed at the end of each test
+     * <p>
+     * Important in case tests end early due to an exception or failure.
+     */
+    @AfterTest
+    public void completeWaitingFutures() {
+        waitingFutures.forEach((future) -> {
+            future.complete(null);
+        });
+        waitingFutures.clear();
+    }
+
+    /**
+     * Use this method to obtain futures for passing to methods on
+     * {@link AsyncClient}
+     * <p>
+     * Using this factory method ensures they will be completed at the end of
+     * the test if your test fails.
+     */
+    private CompletableFuture<Void> newWaitingFuture() {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        waitingFutures.add(result);
+        return result;
+    }
+
+    /**
+     * A helper method to complete a waiting future with a more readable syntax
+     */
+    private void complete(CompletableFuture<?> future) {
+        future.complete(null);
+    }
+
+    private void handleCompletionException(CompletionException e) throws AssertionError {
+        throw new AssertionError("testAsync: unexpected Exception calling service: "
+                + e.getCause().getMessage());
```

Comment:
If you're going to catch the exception and throw an `AssertionError`, can you include the completion exception as the cause so that the test results include the full stack trace?

Alternatively, just let the `CompletionException` be thrown with no special handling.